packages/experimental/ActivityIndicator/src/ActivityIndicator.tsx

```diff
@@ -0,0 +1,95 @@
+import * as React from 'react';
+import { useRef } from 'react';
+import { View, Animated, Easing, ViewProps } from 'react-native';
+
+/**
+ * let R = big circle radius, from middle of circle to middle of border width = (width - borderRadius) / 2
+ * borderRadius only refers to big circle radius
+ * D = sqrt(2)/2 * R is distance from center to the middle of end circle
+ * Top end circle:
+ *   top: R - D - borderRadius
+ *     top edge is already 1 borderRadius below the big circle
+ *   left: R - D - borderRadius
+ *     left edge is already 1 borderRadius to the right
+ * Bottom end circle:
+ *   top: R + D - 2 * borderRadius
+ *     1 borderRadius explained same as top end circle.
+ *     Extra borderRadius since bottom end circle is 1 borderRadius below top end circle already.
+ *   left: R - D - borderRadius
+ *     same explanation as above
+ */
+// This will be a token later
+// radius is from center to halfway border, border is line thickness
+export const getActivityIndicatorStyle = (radius: number, border: number) => {
+  const xydist = Math.sqrt(2)/2 * radius;
+
+  return {
+    semicircle: {
+      width: radius*2 + border, // from outside edge of circle on one side to the opposite side
+      height: radius*2 + border,
+      borderWidth: border,
+      borderColor: '#919191',
+      borderRadius: radius + border/2,
+      borderLeftColor: 'transparent',
+    },
+    endTop: {
+      width: border,
+      height: border,
+      borderRadius: border/2,
+      backgroundColor: '#919191',
+      top: radius - xydist - border,
+      left: radius - xydist - border,
+    },
+    endBottom: {
+      width: border,
+      height: border,
+      borderRadius: border/2,
+      backgroundColor: '#919191',
+      top: radius + xydist - 2*border,
+      left: radius - xydist - border,
+    }
+  }
+}
+
+
+export type ActivityIndicatorProps = ViewProps & {
+    animating?: boolean;
+    hidesWhenStopped?: boolean;
+}
+export const ActivityIndicator = (props: ActivityIndicatorProps) => {
+  const spinAnimation = useRef(new Animated.Value(0)).current;
+  if (props.animating) {
+    React.useEffect(() => {
+      Animated.loop(
+          Animated.timing(spinAnimation, {
+            toValue: 359,
+            duration: 750,
+            useNativeDriver: true,
+            easing: Easing.linear
+          })
+      ).start();
+    });
+  }
```

Comment:
Something about defining a hook only on a boolean feels wrong to me... but I'm not sure if it's actually a problem.
For future reference, could you link the tutorial /documentaion where you learned how to do this in case we need to go back and revisit this later?

Reply:
This is where I got the useEffect and loop part: https://reactnativeforyou.com/react-native-animation-tutorials-how-to-create-looping-animations-in-react-native/

Reply:
Looking at the [Rules of Hooks](https://reactjs.org/docs/hooks-rules.html#explanation), it seems we are violating the "Don't put hooks inside conditionals" part. Their suggestion is to put the condition inside the hook itself. 
Hooks rely on their order of calls matching between different renders, which is why conditions stopping them from being called messes with them. 

Another alternative could be to do what Callstack does and have separate start/stop animation methods https://github.com/callstack/react-native-paper/blob/1638b94a0fd8a481dfa90bf5833d887023eaec5d/src/components/ActivityIndicator.tsx#L83
I think they need that because their animation is much more complicated though.

Reply:
Yeah, after reading that, it makes sense and I put the conditional inside the `.useEffect()`. Even with Callstack, although they have their animations defined outside, the conditionals for which one eventually gets called is within `.useEffect()` as well.